### PR TITLE
Pmp test

### PR DESCRIPTION
**Pull request by silabs-kun**
Completed tests:
DefaultNone, DefaultFull, ResetRegisters, MmodeOnly.
Ongoing tests:
NapotMatching, TorMatching (assertion error), TorZero, TorNomatch.

### CODE REVIEW

File: cv32e40s/tests/cfg/pmp.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: pmp
+description: Default pmp configuration for CV32E40S simulations
+compile_flags: >
+  +define+PMP_ENABLE_64
+ovpsim: >
+  --override root/cpu/PMP_registers=64
+  --showoverrides
+cflags: >
+plusargs: >
+cv_sw_march: rv32imc
```
Comment (silabs-robin) [inline cv32e40s/tests/cfg/pmp.yaml:10]: I am unsure about this `cv_sw_march` and the `cflags` and `plusargs` above it.
I'll think about it a bit more and discuss with others.
Leaving this comment for now, and will come back to it later.

File: cv32e40s/tests/programs/custom/pmp/pmp.c [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// #include <stdio.h>
+// #include <stdlib.h>
```
Comment (silabs-robin) [inline cv32e40s/tests/programs/custom/pmp/pmp.c:2]: If these includes are not needed, might as well remove the comments.

Also, more importantly, we need license headers on all files.
Have a look at other tests and see how license headers are done.

**Comment by MikeOpenHWGroup** [comment]
Hi @silabs-kun, congratulations on your first pull-request to core-v-verif!  I have assigned Henrik to review, and I also have a couple of "administrative" comments:
1. All new source files must have a license header at the top of each file:
```
// SPDX-License-Identifier: Apache-2.0 WITH SHL-2.1
```
2. This is a relatively complex test-program and it would be useful to add a `README.md` to document the goals and theory-of-operation of the test-program.

File: cv32e40s/tests/programs/custom/pmp/pmp.c [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+#include "pmp.h"
+
+int main(int argc, char *argv[])
+{
+  // reset_registers();
+  // default_full();
+  // default_none();
+  // mmode_only();
+  // napot_matching();
+  tor_macthing();
```
Comment (silabs-robin) [inline cv32e40s/tests/programs/custom/pmp/pmp.c:13]: We need a way to run all tests with "push-button" ease.
All of those tests should be runnable without having to edit the source files.

**Comment by silabs-kun** [comment]
> Hi @silabs-kun, congratulations on your first pull-request to core-v-verif! I have assigned Henrik to review, and I also have a couple of "administrative" comments:
> 
> 1. All new source files must have a license header at the top of each file:
> 
> ```
> // SPDX-License-Identifier: Apache-2.0 WITH SHL-2.1
> ```
> 
> 2. This is a relatively complex test-program and it would be useful to add a `README.md` to document the goals and theory-of-operation of the test-program.

Hi @MikeOpenHWGroup thank you for the comment, will add  license header.

File: cv32e40s/tests/programs/custom/pmp/pmp.h [hunk 4]
```diff
@@ -0,0 +1,67 @@
+
+#ifndef PMP_H
+#define PMP_H
+
+// #define CODE_SECTION ".text"
+// #define CODE __attribute__((section(CODE_SECTION)))
```
Comment (silabs-robin) [inline cv32e40s/tests/programs/custom/pmp/pmp.h:6]: Please clean up any commented-out code that is not used.

File: cv32e40s/tests/programs/custom/pmp/temp.c [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#include "pmp.h"
+
+// void temp()
```
Comment (silabs-robin) [inline cv32e40s/tests/programs/custom/pmp/temp.c:3]: Is this `temp.c` file unused?

**Comment by silabs-hfegran** [comment]
Can we change this to a draft PR until it is ready for proper review?

**Pull request closed without merging**